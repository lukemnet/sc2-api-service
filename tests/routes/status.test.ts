import fastify from 'fastify';
import server from '../../src/index';
import getConfig from '../helper';

describe('/status (Redis enabled)', () => {
  const fastifyServer = fastify();

  beforeAll(() => {
    fastifyServer.register(server, getConfig(true));
  });

  afterEach(() => {
    fastifyServer.close();
  });

  
  it('returns 200', async () => {
    const res = await fastifyServer.inject({ method: 'GET', url: '/status' });
    expect(res.statusCode).toBe(200);
  });

  it('returns correct response', async () => {
    const res = await fastifyServer.inject({ method: 'GET', url: '/status' });
    expect(res.payload).toMatchSnapshot();
  });
});

describe('/status (Redis disabled)', () => {
  const fastifyServer = fastify();

  beforeAll(() => {
    fastifyServer.register(server, getConfig(false));
  });

  afterEach(() => {
    fastifyServer.close();
  });

  
  it('returns 200', async () => {
    const res = await fastifyServer.inject({ method: 'GET', url: '/status' });
    expect(res.statusCode).toBe(200);
  });

  it('returns correct response', async () => {
    const res = await fastifyServer.inject({ method: 'GET', url: '/status' });
    expect(res.payload).toMatchSnapshot();
  });
});