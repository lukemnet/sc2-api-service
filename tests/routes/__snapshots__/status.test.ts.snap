// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/status (Redis disabled) returns correct response 1`] = `"{\\"status\\":200,\\"message\\":\\"ok\\"}"`;

exports[`/status (Redis enabled) returns correct response 1`] = `"{\\"status\\":200,\\"message\\":\\"ok\\"}"`;
